aana/deployments/vllm_deployment.py

```diff
@@ -32,30 +33,30 @@ class VLLMConfig(BaseModel):
     """The configuration of the vLLM deployment.
 
     Attributes:
-        model (str): The model name.
-        dtype (str): The data type. Defaults to "auto".
-        quantization (str): The quantization method. Defaults to None.
+        model_id (str): The model name.
+        dtype (Dtype): The data type. Defaults to Dtype.AUTO.
+        quantization (str | None): The quantization method. Defaults to None.
         gpu_memory_reserved (float): The GPU memory reserved for the model in MB.
         default_sampling_params (SamplingParams): The default sampling parameters.
             Defaults to SamplingParams(temperature=0, max_tokens=256).
-        max_model_len (int): The maximum generated text length in tokens. Defaults to None.
-        chat_template (str): The name of the chat template. If not provided, the chat template
+        max_model_len (int | None): The maximum generated text length in tokens. Defaults to None.
+        chat_template (str | None): The name of the chat template. If not provided, the chat template
             from the model will be used. Some models may not have a chat template.
             Defaults to None.
         enforce_eager (bool): Whether to enforce eager execution. Defaults to False.
         engine_args (CustomConfig): Extra engine arguments. Defaults to {}.
     """
 
-    model: str
-    dtype: str | None = Field(default="auto")
+    model_id: str = Field(validation_alias=AliasChoices("model_id", "model"))
+    dtype: Dtype = Field(default=Dtype.AUTO)
     quantization: str | None = Field(default=None)
     gpu_memory_reserved: float
     default_sampling_params: SamplingParams = SamplingParams(
         temperature=0, max_tokens=256
     )
     max_model_len: int | None = Field(default=None)
     chat_template: str | None = Field(default=None)
-    enforce_eager: bool | None = Field(default=False)
+    enforce_eager: bool = Field(default=False)
```

Comment:
Check this out apperently if ur a tiny bit short on GPU resources u can set this to true and save a bit of memory. https://pytorch.org/blog/accelerating-pytorch-with-cuda-graphs/
https://github.com/vllm-project/vllm/issues/4449

Reply:
I know. That's why we added it in the first place. I also mentioned it in the docs: https://github.com/mobiusml/aana_sdk/blob/2202beaaf7caee66e6f6c7d610cb8e69aa8db145/docs/pages/model_hub/text_generation.md?plain=1#L49